src/SelfService/Application/KafkaTopicApplicationService.cs

```diff
@@ -24,6 +26,205 @@ IKafkaTopicRepository kafkaTopicRepository
         _kafkaTopicRepository = kafkaTopicRepository;
     }
 
+    public async Task CheckIfCanRequestContract(
+        KafkaTopicId kafkaTopicId,
+        MessageType messageType,
+        MessageContractSchema newSchema
+    )
+    {
+        var schemaVersion = newSchema.GetSchemaVersion();
+        if (schemaVersion == null)
+            throw new ArgumentException("Cannot request new message contract without schema version");
+
+        var contracts = (await _messageContractRepository.FindBy(kafkaTopicId, messageType)).ToArray();
+        if (contracts.Length == 0)
+        {
+            if (schemaVersion != 1)
+                throw new ArgumentException("Cannot request new message contract with schema version other than 1");
+            return;
+        }
+
+        var latestSchemaVersion = 0;
+        MessageContract? latestContract = null;
+        foreach (MessageContract contract in contracts)
```

Comment:
Instead of first fetching all contracts, checking length, and then looping over them to find the latest contract; Would it make sense to repository-method allowing lookup of the oldest contract as `Option[MessageContract]`.

It would save a fair few lines in this function at the small cost of adding fewer lines somewhere else.

Reply:
rewrote to get latest schema